flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/source/split/JdbcSourceSplitSerializer.java

```diff
@@ -106,7 +106,7 @@ public JdbcSourceSplit deserializeJdbcSourceSplit(DataInputStream in)
         byte[] chkOffsetBytes = new byte[chkOffsetBytesLen];
         in.read(chkOffsetBytes);
         CheckpointedOffset chkOffset =
-                InstantiationUtil.deserializeObject(chkOffsetBytes, in.getClass().getClassLoader());
+                InstantiationUtil.deserializeObject(chkOffsetBytes, CheckpointedOffset.class.getClassLoader());
```

Comment:
Thanks @jan-osc for the pr.
Would you like to add a test case for the fix ?

Reply:
Hello. This bug only appears when the JDBC connector is deployed to the cluster during the deserialization of the `JdbcSourceSplit` from the state. I'll think through how I could simulate these conditions in tests and if it's really needed. In the simplest test case, we could just test if serialization/deserialization works but with a JUnit class loader (flat class loader). Class loader setup in Flink is more complicated (FlinkUserCodeClassLoader).